uv.lock

```diff
@@ -0,0 +1,1024 @@
+version = 1
```

Comment:
do we need to check this file in?

Reply:
I'm not sure on this one to be fair, as heroku doesn't support uv by default, we need a preprocessor to generate requirements.txt from uv.lock [here](https://github.com/dropseed/heroku-buildpack-uv?tab=readme-ov-file)

I remember from when I used poetry, having .lock in VCS was the correct practice when producing an executable, not a library.